python/ray/train/v2/xgboost/xgboost_trainer.py

```diff
@@ -67,7 +67,7 @@ def train_fn_per_worker(config: dict):
         train_ds = ray.data.from_items([{"x": x, "y": x + 1} for x in range(32)])
         eval_ds = ray.data.from_items([{"x": x, "y": x + 1} for x in range(16)])
         trainer = XGBoostTrainer(
-            train_fn_per_worker,
+            train_loop_per_worker=train_fn_per_worker,
```

Comment:
In the long term this is the main API change, right? That this needs to be a kwarg.

Reply:
Actually can we just keep this as a required argument since the V1 will always populate it? Are there any problems if we do?

Reply:
The problem is that this must be optional in order for the V1 API to be able to take in 2 different inputs.

See here: https://github.com/ray-project/ray/pull/50045/files/8ac15ae8ba595d5af49f98808f38cb8b90a27ac3#diff-a991023b88ffebf906bfc5b480e1004257e2b65aa075c2e1edd5a331cf2fe836R191

Reply:
This is different from the `DataParallelTrainer` API which does enforce a non kwarg `train_loop_per_worker`.

However, if we were to keep the non-kwarg required param, usage of the legacy params would require passing `None` as the first param.

```python
XGBoostTrainer(None, datasets={...}, params={...})
```

Reply:
Nvm figured out a solution.